build/utils-xpc.js

```diff
@@ -1269,8 +1269,33 @@ function NodeHelper() {
     var node = new Commander('node');
     node.initPath(getEnvPath());
     this.require = function(path, options) {
-      node.run(['--harmony', '-e', 'require("./build/' + path + '").execute(' +
-        JSON.stringify(options) + ')']);
+      var result = '';
+      var done = false;
+      node.runWithSubprocess(['--harmony', '-e',
+        'require("./build/' + path + '").execute(' +
+        JSON.stringify(options) + ')'], {
+          stdout: function(data) {
+            result += data;
+            dump(data);
+          },
+          stderr: function(err) {
+            dump(err);
+          },
+          done: function() {
+            done = true;
+          }
+        });
+
+      processEvents(function() {
+        return {
+          wait: !done
+        };
+      });
```

Comment:
I assume ```runWithSubprocess``` is an async so let's set ``processEvents()`` to block main thread until it finishes.